### ISSUE TITLE
sqlite_sequence error

### ISSUE THREAD

**Issue by airstep** [dpa99c/cordova-sqlite-porter#22]
By documentation you [can't drop this table](https://sqlite.org/fileformat2.html#seqtab) (sqlite_sequence) on import:
message:"Failed to import SQL; message=sqlite3_prepare_v2 failure: table sqlite_sequence may not be dropped"

So... need to remove from exported sql:
```
DROP TABLE IF EXISTS `sqlite_sequence`;
CREATE TABLE sqlite_sequence(name,seq);
```

**Comment by dpa99c** [comment]
On what platform version are you seeing the `sqlite_sequence` table being included in the SQL data dump?
Using WebSQL DB or native SQLIte?

**Comment by airstep** [comment]
native SQLIte

**Comment by imaadhrizni** [comment]
Hi @dpa99c . 
This issue is only occurring for me on native SQLite. I'm using this plugin with Ionic2.

I've come across a similar dilemma where the `sqlite_sequence` table is being included in the JSON export. So when it's being re-imported the sequences are taken first from the JSON and then recreated as well again for each table that has a UNIQUE AUTO INCREMENT key. Is there any way to bypass the `sqlite_squence` table being exported on native SQLite? 

Thanks and regards,
Imaadh

**Comment by dpa99c** [comment]
Should be resolved by the above commit, which is published to npm as v0.1.6.

**Comment by dpa99c** [comment]
In fact, try v0.1.7 which makes sure, in addition to excluding the DROP statements from exported SQL, CREATE statements are also excluded.

**Comment by imaadhrizni** [comment]
Hi @dpa99c .
I'm not sure how to check it through my current implementation as I'm using this plugin through @ionicnative (https://ionicframework.com/docs/native/sqlite-porter/). That versioning is different to the original version in npm (which is 0.1.7) while on ionic native it goes as (v3.12.1).

**Comment by dpa99c** [comment]
Ionic Native is only a Typescript wrapper around the plugin code, which you still need to install into your project. Hence you can specify the version:

    ionic cordova plugin add uk.co.workingedge.cordova.plugin.sqliteporter@0.1.7

**Comment by imaadhrizni** [comment]
Hi Dave,
I installed it as you suggested(thanks for the tip :)). When I took the database dump this is the json being dumped : 
```
{
   "structure":{
      "tables":{
         "data":"(ID INTEGER PRIMARY KEY AUTOINCREMENT, data VARCHAR(32))"
      },
      "otherSQL":[

      ]
   },
   "data":{
      "inserts":{
         "data":[
            {
               "ID":1,
               "data":"Abc"
            },
            {
               "ID":2,
               "data":"Def"
            }
         ],
         "sqlite_sequence":[
            {
               "name":"data",
               "seq":2
            }
         ]
      }
   }
}
```

Is this correct? I can see in your commit that you've explicitly checked for all reserved tables and not included them in the dump. Although the structure is not included the data is being included.

Sorry if I'm incorrect in anyway, I'm a bit naive in Cordova.

**Comment by dpa99c** [comment]
This is published in v1.0.0 - the above commit stops the reserved table data being exported in the data structure, as @imaadhrizni observed.

Let me know if you still have issues with v1.0.0
